numpy/core/_methods.py

```diff
@@ -54,20 +54,29 @@ def _mean(a, axis=None, dtype=None, out=None, keepdims=False):
     arr = asanyarray(a)
 
     rcount = _count_reduce_items(arr, axis)
+    orig_dtype = dtype
```

Comment:
A comment as to intent would be helpful here. I'd also move this up above the previous line.